Base/QTCore/qSlicerCoreApplication.cxx

```diff
@@ -353,6 +354,12 @@ void qSlicerCoreApplicationPrivate::init()
     vtkEventBroker::GetInstance()->SetRequestModifiedCallback(modifiedRequestCallback);
   }
 
+  // Create the translation object used for string translation in logic classes
+  if (!this->AppLogic->GetTranslatorInstance())
+  {
+    this->AppLogic->SetTranslatorInstance(new qMRMLTranslator());
```

Comment:
qMRMLTranslator is not a Qt class, and it is not taking care of destroying itself if it is created by new. Instead, it is a VTK class, which can be created for example by vtkNew:

```python
vtkNew<vtkMRMLTranslator> translator;
this->AppLogic->SetTranslatorInstance(translator);
```